request.go

```diff
@@ -99,3 +100,24 @@ func (req *Request) getKey() (string, bool) {
 
 	return *req.key, true
 }
+
+func parseKey(k string) string {
+	var key string
+	var s bool
+	for i := 0; i < len(k); i++ {
+		switch string(k[i]) {
+		case "{":
+			s = true
+		case "}":
+			break
+		default:
+			if s {
+				key += string(k[i])
+			}
+		}
+	}
+	if key == "" {
+		return k
+	}
+	return key
+}
```

Comment:
One way to implement this more efficiently would be to use the strings.IndexByte function and return a subslice of the original string, do you want to give it a shot?